src/robotFactory.js

```diff
@@ -1,10 +1,110 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords = {
+      x: 0,
+      y: 0,
+    },
+    chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = coords;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    if (this.coords.x === undefined) {
+      this.coords.x = 0;
+    }
 
-class DeliveryDrone {}
+    if (this.coords.y === undefined) {
+      this.coords.y = 0;
+    }
+  }
+
+  goForward(step = 1) {
+    this.coords.y += step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  }
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, \
+Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(
+    name,
+    weight,
+    coords = {
+      x: 0,
+      y: 0,
+      z: 0,
+    },
+    chipVersion) {
+    super(name,
+      weight,
+      coords,
+      chipVersion);
+
+    if (this.coords.z === undefined) {
+      this.coords.z = 0;
+    }
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  }
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(
+    name,
+    weight,
+    coords = {
+      x: 0,
+      y: 0,
+      z: 0,
+    },
```

Comment:
```suggestion
    coords = {},
```
We don't need it here because it takes cords from parent (Flying robot).